java/core/lance-jni/src/lib.rs

```diff
@@ -88,3 +99,178 @@ pub extern "system" fn Java_com_lancedb_lance_JniLoader_initLanceLogger() {
     log::set_max_level(max_level);
     // todo: add tracing
 }
+
+#[no_mangle]
+pub extern "system" fn Java_com_lancedb_lance_MergeInsertBuilder_createNativeBuilder(
+    mut env: JNIEnv,
+    _class: JClass,
+    dataset_handle: jlong,
+    on_columns: JObjectArray,
+) -> jlong {
+    let dataset = unsafe { &*(dataset_handle as *const LanceDataset) };
+
+    // 转换 Java String 数组为 Rust Vec<String>
```

Comment:
Please avoid Chinese comments. By the way, I think we don't need this comment at all.